app/workers/send_email_worker.rb

```diff
@@ -4,13 +4,21 @@ class SendEmailWorker < ApplicationWorker
   RATE_LIMIT_THRESHOLD = 21_600 # max requests in a minute, equates to 350 a second
   RATE_LIMIT_INTERVAL = 60
 
-  sidekiq_options retry: 9
-
   sidekiq_retries_exhausted do |msg|
     Email.find_by(id: msg["args"].first, status: :pending)
          &.update!(status: :failed)
   end
 
+  sidekiq_options lock: :until_executed,
+                  unique_across_queues: true,
```

Comment:
Note to self: [the doc for this option](https://github.com/mhenrixon/sidekiq-unique-jobs#unique_across_queues) is confusing, but [the actual code makes sense](https://github.com/mhenrixon/sidekiq-unique-jobs/blob/fb45ea748732310ae9299dd60820b36b31d8bc18/lib/sidekiq_unique_jobs/lock_digest.rb#L62), and also clarifies that the class is still considered as part of the uniqueness.

Reply:
Yeah the doc isn't clear at all (apparently their documentation is being overhauled in version 7). 

Hopefully the commit message and specs give a better indication of its use. 😄